### DISCUSSION TITLE
Photosync - 413 - request too large

### DISCUSSION THREAD

**Discussion by veritasecops** [photoprism/photoprism#3459] [General]
How can I adjust webdav settings to accept larger files? I am getting this error on Photosync.

**Comment by lastzero** [comment]
Using NGINX? 👉 https://docs.photoprism.app/getting-started/proxies/nginx/

**Reply by veritasecops** [reply]
Thanks for the reply!

**Comment by veritasecops** [comment]
I have resolved this. I adjusted my NGINX proxy to be able to manage web sockets appropriately, and added the directive 'client_max_body_size 1000M;' to the NGINX configuration file under the server -> location node.

**Reply by lal309** [reply]
Can you share a screenshot or copy/paste your solution? I'm running PhotoPrism behind NGINX Proxy Manager on Unraid as docker containers but I can't seem to solve this issue.

**Reply by veritasecops** [reply]
`server {
	server_name photoprism.lan;
	location / {
		proxy_pass http://localhost:2342;
		}
}

server {
	server_name <your fqdn>;
	client_max_body_size 1000M; #THIS IS THE LINE I CHANGED

	location / {
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header Host $host;
		proxy_pass http://localhost:2342;
		proxy_buffering off;
		proxy_http_version 1.1;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection "upgrade";
		}


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/<your fqdn>/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/<your fqdn>/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}


server {
    if ($host = <your fqdn>) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	server_name <your fqdn>;
    listen 80;
    return 404; # managed by Certbot


}`

**Comment by veritasecops** [comment]
`server {
	server_name photoprism.lan;
	location / {
		proxy_pass http://localhost:2342;
		}
}

server {
	server_name <your fqdn>;
	client_max_body_size 1000M; #THIS IS THE LINE I CHANGED

	location / {
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header Host $host;
		proxy_pass http://localhost:2342;
		proxy_buffering off;
		proxy_http_version 1.1;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection "upgrade";
		}


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/<your fqdn>/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/<your fqdn>/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}


server {
    if ($host = <your fqdn>) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	server_name <your fqdn>;
    listen 80;
    return 404; # managed by Certbot


}`

**Comment by maxiride** [comment]
This thread seems the only occurrence when searching for the error "photoprism 413" so I'm chiming in.

Photoprism CE Build 231021-9abea5b55
Caddy v2

I'm using PhotoSync on mobile to push photos and videos to Photoprism, the issue occurs when pushing videos weighting more than 10 MB (approximate guess based on when the error occurs, I tried to annotate the file weight).

Caddyfile exactly as in the [documentation](https://docs.photoprism.app/getting-started/proxies/caddy-2/), all the threads on the Caddy forum points to issues with the upstream service so I'm looking for direction to investigate further the issue or provide more logs where necessary.

**Reply by lastzero** [reply]
You mean error 413 is sent by PhotoPrism (the upstream service) and Caddy just forwards it? I am not aware that we have implemented it in our application, and if there was a general upload limit of 10 MB in PhotoPrism, it is highly unlikely that no one else has discovered this yet...

**Reply by maxiride** [reply]
Yeah, that's odd indeed. Every topic I can find on the Caddy forums reports an upstream issue. But I'll try to open a thread there too and see if anything new arises.

**Comment by forest-cat** [comment]
If youre using nextcloud you should look into settings for your nextcloud too, nextclouds default upload limit is at 512M

**Comment by tturnerdev** [comment]
Similar to @maxiride, this came up when searching _PhotoSync 413 error_.

I'm transferring to PhotoPrism and it is installed as an app on TrueNAS Scale (older version that still used k8s with Traefik reverse proxy ingress.)

The 413 error definitely seems to be caused by the reverse proxy, as when I change the PhotoPrism config to go direct to the app/port on TrueNAS by using the server's IP address (instead of app domain name, e.g. photos.example.com, which uses Traefik reverse proxy ingress) and the app/container's exposed port (e.g. 2342) with the resulting URL in PhotoSync's PhotoPrism config like: `http://192.168.0.123:2342`, then I no longer get the error.

I'll use this direct URL for now, but I'm sure I can fix the Traefik config for the app to increase the `client_max_body_size` just like is mentioned above for nginx reverse proxy. I don't have time to try that or troubleshoot further, but I'm confident that would also resolve the problem here.

**Reply by maxiride** [reply]
I later discovered the issue was that my server was behind Claudflare protection (orange cloud ON) and the free tier has a limit of 100 MB per request.
